src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -288,184 +268,255 @@ public void postAuthnLogin(String userAgent, String xForwardedFor,
             .respond400WithTextPlain("You must provide a password")));
         return;
       }
-      if (entity.getUserId() != null && !requireActiveUser) {
-        logger.debug("No need to look up user id");
-        userVerified = Future.succeededFuture(new JsonObject()
-            .put("id", entity.getUserId()).put(ACTIVE, true)
-            .put(USERNAME, "__undefined__"));
-      } else {
-        logger.debug("Need to look up user id");
-        if (entity.getUserId() != null) {
-          userVerified = lookupUser(null, entity.getUserId(), tenantId, okapiURL,
-              requestToken);
-        } else {
-          userVerified = lookupUser(entity.getUsername(), null, tenantId, okapiURL,
-              requestToken);
-        }
-      }
-      userVerified.onComplete(verifyResult -> {
-        if (verifyResult.failed()) {
-          String errMsg = "Error verifying user existence: " + verifyResult
-              .cause().getLocalizedMessage();
-          logger.error(errMsg);
-          asyncResultHandler.handle(Future.succeededFuture(
-              PostAuthnLoginResponse.respond422WithApplicationJson(
-                  getErrors(errMsg, CODE_USERNAME_INCORRECT, new ImmutablePair<>(USERNAME, entity.getUsername())))
-          ));
+      handleCrossTenantLogin(entity, tenantId, okapiHeaders, asyncResultHandler)
```

Comment:
Even regular logins will execute a method called `handleCrossTenantLogin`? Reads like it shouldn't be the case.

Reply:
Yes, Serhii suggested always calling mod-users' /user-tenants and calling it cross-tenant rather than consortia. What would you prefer ?

Reply:
ok, i sorta get it now! could it be handleLogin?

Reply:
Renamed